src/classes/PS_GatewayManagement_TEST.cls

```diff
@@ -0,0 +1,129 @@
+ *     BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *     CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *     LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *     ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *     POSSIBILITY OF SUCH DAMAGE.
+ */
+
+@isTest
+private with sharing class PS_GatewayManagement_TEST {
+
+    private static Map<String, List<Map<String, String>>> configParams;
+    private static Map<String, PS_IntegrationService.Config> paymentConfigParams;
+    private static Map<String, PS_IntegrationService.Config> makanaConfigParams;
+    private static final String API_KEY = 'apiKey';
+    private static final String PAYMENTS_APPLICATION_TYPE = 'payments';
+
+    public static final String TEST_MERCHANT_ID = 'abc123-xyz456';
+    public static final String TEST_GATEWAY_ID = 'def123-ghi456';
+    public static final String TEST_API_KEY = 'def1234-tuv4567';
+    public static final String TEST_BASE_URL = 'https://elevate.test.com';
+
+    // Key-Values by service name
+    public static final Map<String, List<Map<String, String>>> testConfig = new Map<String, List<Map<String, String>>>{
+        PS_IntegrationServiceConfig.PAYMENTS_SERVICE_NAME =>
+            new List<Map<String, String>>{
+                new Map<String, String>{
+                        'key' => PS_IntegrationServiceConfig.SFDO_MERCHANTIDS,
+                        'value' => TEST_MERCHANT_ID
+                },
+                new Map<String, String>{
+                        'key' => PS_IntegrationServiceConfig.SFDO_GATEWAYIDS,
+                        'value' => TEST_GATEWAY_ID
+                },
+                new Map<String, String>{
+                        'key' => PS_IntegrationServiceConfig.API_KEY,
+                        'value' => TEST_API_KEY
+                },
+                new Map<String, String>{
+                        'key' => PS_IntegrationServiceConfig.BASE_URL,
+                        'value' => TEST_BASE_URL
+                }
+        }
+    };
+
+    @isTest
+    static void shouldUpdateGatewayId() {
+        Test.startTest();
+
+        PS_GatewayManagement.setGatewayId(TEST_GATEWAY_ID);
+
+        Test.stopTest();
+
+        System.assertEquals(TEST_GATEWAY_ID, new PS_IntegrationServiceConfig.Service().getGatewayIds(), 'The gateway ' +
+                'id stored in CRM should be successfully updated to the new value provided.');
+    }
+
+    @isTest
+    static void shouldRetrieveCorrectGatewayId() {
+        Test.startTest();
+
+        PS_GatewayManagement.setGatewayId(TEST_GATEWAY_ID);
+
+        Test.stopTest();
+
+        System.assertEquals(TEST_GATEWAY_ID, PS_GatewayManagement.getGatewayIdFromConfig(), 'When this method is ' +
+                'invoked, the gateway Id stored in CRM should be the gateway returned.');
+    }
```

Comment:
It looks like `PS_GatewayManagement.getGatewayIdFromConfig()` calls `PS_IntegrationServiceConfig.Service().getGatewayIds()` within its method body.  If that is the case, could one of these tests be removed?

Reply:
@kenlewis-sfdo I think at this point we'll avoid any further code changes unless we feel they are necessary.